include/picongpu/particles/manipulators/unary/FreeTotalCellOffset.hpp

```diff
@@ -0,0 +1,160 @@
+
+#include <string>
+
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+namespace unary
+{
+namespace acc
+{
+    template< typename T_Functor >
+    struct FreeTotalCellOffset : private T_Functor
+    {
+
+        using Functor = T_Functor;
+
+        HDINLINE FreeTotalCellOffset(
+            Functor const & functor,
+            DataSpace< simDim > const & superCellToLocalOriginCellOffset
+        ) :
+            T_Functor( functor ),
+            m_superCellToLocalOriginCellOffset( superCellToLocalOriginCellOffset )
+        {
+        }
+
+        /** call user functor
+         *
+         * The random number generator is initialized with the first call.
+         *
+         * @tparam T_Particle type of the particle to manipulate
+         * @tparam T_Args type of the arguments passed to the user functor
+         * @tparam T_Acc alpaka accelerator type
+         *
+         * @param alpaka accelerator
+         * @param particle particle which is given to the user functor
+         * @return void is used to enable the operator if the user functor except two arguments
+         */
+        template<
+            typename T_Particle,
+            typename T_Acc
+        >
+        HDINLINE
+        void operator()(
+            T_Acc const &,
+            T_Particle & particle
+        )
+        {
+            DataSpace< simDim > const cellInSuperCell(
+                DataSpaceOperations< simDim >::template map< SuperCellSize >( particle[ localCellIdx_ ] )
+            );
+            Functor::operator( )(
+                m_superCellToLocalOriginCellOffset + cellInSuperCell,
+                particle
+            );
+        }
+
+    private:
+
+        DataSpace< simDim > const m_superCellToLocalOriginCellOffset;
+    };
+} // namespace acc
+
+    template< typename T_Functor >
+    struct FreeTotalCellOffset :
+        protected functor::User< T_Functor >,
+        private functor::misc::TotalCellOffset
+    {
+        using CellOffsetFunctor = functor::misc::TotalCellOffset;
+        using Functor = functor::User< T_Functor >;
+
+        template< typename T_SpeciesType >
+        struct apply
+        {
+            using type = FreeTotalCellOffset;
+        };
+
+        /** constructor
+         *
+         * @param currentStep current simulation time step
+         */
+        HINLINE FreeTotalCellOffset( uint32_t currentStep ) :
+            Functor( currentStep ),
+            CellOffsetFunctor( currentStep )
+        {
+        }
+
+        /** create functor for the accelerator
+         *
+         * @tparam T_WorkerCfg pmacc::mappings::threads::WorkerCfg, configuration of the worker
+         * @tparam T_Acc alpaka accelerator type
+         *
+         * @param alpaka accelerator
+         * @param localSupercellOffset offset (in superCells, without any guards) relative
+         *                        to the origin of the local domain
```

Comment:
QUESTION:
What does the coding standard says on alignment here?

Reply:
nothing